### ISSUE TITLE
Runtime Error.. ..Nativescript 2.4.2, Typescript 2.1.4

### ISSUE THREAD

**Issue by holymp2006** [bradmartin/nativescript-floatingactionbutton#43]
I installed the plugin, but after build, I tried to run my app and got the following error.

```
JS ERROR Error: Building UI from XML. @file:///app/views/verify/verify.xml:21:21
	 > Module '/Users/sam/Library/Developer/CoreSimulator/Devices/8FEEE46E-D4AB-41D8-AD5D-7F4AD5F5A412/data/Containers/Bundle/Application/BEA6CB80-A2F8-4999-86A7-4814D12505C4/xxxxx.app/app/nativescript-floatingactionbutton' not found for element 'nativescript-floatingactionbutton:Fab'.
	   > Could not find module '/Users/sam/Library/Developer/CoreSimulator/Devices/8FEEE46E-D4AB-41D8-AD5D-7F4AD5F5A412/data/Containers/Bundle/Application/BEA6CB80-A2F8-4999-86A7-4814D12505C4/xxxxx.app/app/nativescript-floatingactionbutton'. Computed path '/Users/sam/Library/Developer/CoreSimulator/Devices/8FEEE46E-D4AB-41D8-AD5D-7F4AD5F5A412/data/Containers/Bundle/Application/BEA6CB80-A2F8-4999-86A7-4814D12505C4/xxxxx.app/app/nativescript-floatingactionbutton'.
```

my (Typescript) project uses Nativescript 2.4.2 and Typescript 2.1.4.
Is this plugin supported for typescript projects?
I see angular users doing "`registerElement`".

**Comment by bradmartin** [comment]
Yep. Almost any plugin that is plain NATIVESCRIPT works with Angular. Make sure you run a full build after adding plugins with native deps and if using angular you'll need to use `registerElement()` like you mentioned.

**Comment by holymp2006** [comment]
thanks for the quick reply, Brad. I already did multiple full builds before reporting this. I'm not using using angular. my project is just Typescript (that gets "compiled" to js). I don't know if I'm right, but I think a "`.d.ts`" file which contains the module declaration might be needed in the plugin folder. This is obtainable for other plugins I already use.

**Comment by bradmartin** [comment]
Not a requirement. Those are definition/declarations for typescript. I use it all the time. Most of my plugin repos have them. This was the first plugin I did and was in plain js. So having the .d.ts file will help but that's not the issue for anything failing and some IDEs resolve modules better than others. At any rate. I'm fairly certain I have this working with the latest NativeScript version since typescript wouldn't be involved with this plugin at any point. Also if a plugin is in typescript it should really only ship the transpiled .js to npm since TS isnt being ran through anything. 

Sorry for the long message. Have you added the xmlns to the page component in your UI?

**Comment by holymp2006** [comment]
message wasn't long, I read everything. :) ..yes, I added the xmlns to page like you described in usage. 
Just to be sure, below is the xml for the page.

```
<Page loaded="pageLoaded" 
    xmlns="http://schemas.nativescript.org/tns.xsd" 
    xmlns:FAB="nativescript-floatingactionbutton">
    <Page.actionBar>
        <ActionBar title="  "/>
    </Page.actionBar>
    <StackLayout orientation="vertical">
        <GridLayout>
            <GridLayout scaleX="1" scaleY="1" class="background"></GridLayout>
            <GridLayout scaleX="1" scaleY="1" class="background-overlay"></GridLayout>
            <GridLayout class="container" rows="70,auto,auto,*,20">
                <StackLayout row="1">
                    <TextField text="{{ verifyInput }}" id="verifyTextField" hint="xxxxxxx" autocorrect="false" autocapitalizationType="allCharacters" returnPress="verify" returnKeyType="go" />
                    <Button text="Verify" tap="verify" class="btn-primary btn-verify" />
                </StackLayout>
                <StackLayout row="2" verticalAlignment="bottom">
                    <Button text="+" tap="{{ gotoRegister }}" class="register-shortcut" horizontalAlignment="right" /> 
                </StackLayout>
                <StackLayout row="3" verticalAlignment="bottom">
                    <FAB:fab tap="{{ gotoRegister }}"
                     row="1"
                     icon="res://my_icon"
                     rippleColor="#f1f1f1"
                     class="fab-button" />
                </StackLayout>
            </GridLayout>
            <ActivityIndicator busy="{{ isLoading }}" />
        </GridLayout>
    </StackLayout>
</Page>
```

In the code, the `Button` just above your `FAB`, is what I currently use as my FAB. But I couldn't get the border radius to be perfect, so I need yours to work and save me.   :)
I have fully built again, but the problem is still there.

**Comment by bradmartin** [comment]
Yea that's all good. Do me a favor. Create a new app using the CLI. Add the
plugin, change the XML in the template app to the XML on the readme, build
then run the app. Let me know what happens

**Comment by holymp2006** [comment]
ok. I built the new app like you suggested and used your XML. But at startup, the app crashes with the same error (`JS ERROR Error: Building UI from XML. @file:///app/main-page.xml:12:13 ...`) as the one I posted above.

**Comment by bradmartin** [comment]
I just cloned the repo and ran the demo app using the latest NS versions of everything. Can you show me the actual commands you run, in order from starting a new app and trying this plugin? Then we can compare and see where something might be wrong.

**Comment by holymp2006** [comment]
ok.
`tns create my-app-name --template typescript`
I cd into the app project folder and
`tns install typescript`
`npm install nativescript-floatingactionbutton`
paste your page code, change your `loaded="pageLoaded"` to their `loaded="navigatingTo"`, as `navigatingTo` is the exported function in the template's "main-page.ts".
then
`tns build ios`

**Comment by bradmartin** [comment]
K. My flow is:

`tns create test --tsc`
`tns platform add android`
`tns plugin add nativescript-floatingactionbutton`
`tns build android`
`tns run android`

Same build and run for iOS should work. Didn't test while ago though. If
that's causing an issue the cocoapod might not be installing correctly for
you which is another crappy issue with pods and different versions of
cocoapods I've seen.

Try those steps and let me know if it works adding the fab to the page

**Comment by holymp2006** [comment]
Ok, I think I see the problem.
You never said anything about installing as a nativescript plugin.
I installed as an npm package like the instruction says `npm install nativescript-floatingactionbutton
`

`tns install` installs other dependencies too, in this case;
"`Installing MNFloatingActionButton (0.1.5)`”

I installed as nativescript plugin, 
`tns plugin add nativescript-floatingactionbutton`

...and everything works fine.
Thanks for the plugin, Brad. :)
